spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -234,6 +234,13 @@ private static void apply(UpdateSchema pendingUpdate, TableChange.AddColumn add)
         add.isNullable(),
         "Incompatible change: cannot add required column: %s",
         leafName(add.fieldNames()));
+    if (add.defaultValue() != null) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Cannot add column %s since default values are currently unsupported",
```

Comment:
"since setting default values in Spark is currently unsupported?" 

I thought the columns can be read but the code for altering the schema is missing?

Reply:
Yeah they can be read, you're right this error message does leave the impression that it's generally not supported but that's not true. it's specifically that setting default values in Spark is unsupported